programs/fileio.c

```diff
@@ -319,6 +319,8 @@ struct FIO_prefs_s {
     int excludeCompressedFiles;
     int patchFromMode;
     int contentSize;
+    int currFileIdx;
+    int nbFiles;
 };
```

Comment:
out of scope for this PR, but nonetheless impacted by it : 

It seems like `struct FIO_prefs_s` is no longer a "preferences" structure,
and has become a kind of "state" at some point.

The difference is pretty substantial : 
"preferences" (which would probably be better named "parameters") 
implies a set of choices made externally by the calling process 
and which expression shall remain valid during the processing stage.
In essence, once set, it should remain `const` immutable.

By contrast, "state" contains internal (private) fields, often mutated during processing, 
in order to preserve synchronization between functions during a larger processing event,
such as compressing multiple files.

I presume it would be a good idea to separate both concepts more cleanly.
But that's obviously a larger change, for another PR.

Reply:
Yeah, especially if we decide to move forward with the final "status" output that requires tracking compressed bytes throughout the operation - it functions more like a `ZSTD_CCtx`.

Perhaps a followup PR can include a new `FIO_Ctx_s` struct:
```
struct FIO_Ctx_s {
    int nbFiles;
    int currFileIdx;
    int bytesConsumed;
    int bytesEmitted;
};
```

Reply:
I'd be happy to submit a PR for this a bit later.

Reply:
Indeed, good food for thoughts for a potential follow up.

Also, to keep in mind : distinction between public API, with parameters necessarily provided from outside,
and internal state information, which should remain private, hence unaccessible.
The less the external world has to know about, the better (both safer and easier to use).

We'll figure out what to do when it comes to this part.